src/lib/pubkey/kyber/kyber_common/kyber.cpp

```diff
@@ -1225,16 +1281,17 @@ class Kyber_KEM_Encryptor final : public PK_Ops::KEM_Encryption_with_KDF,
 
          // Multitarget countermeasure for coins + contributory KEM
          G->update(shared_secret);
-         G->update(H->process(m_key.public_key_bits_raw()));
+         G->update(m_key.H_public_key_bits_raw());
          const auto g_out = G->final();
 
-         const auto middle = G->output_length() / 2;
-         const auto lower_g_out = secure_vector<uint8_t>(g_out.begin(), g_out.begin() + middle);
-         const auto upper_g_out = secure_vector<uint8_t>(g_out.begin() + middle, g_out.end());
+         BOTAN_ASSERT_EQUAL(g_out.size(), 64, "Expected output length of Kyber G");
+
+         const auto lower_g_out = std::span<const uint8_t>(g_out.data(), 32);
+         const auto upper_g_out = std::span<const uint8_t>(g_out.data() + 32, 32);
```

Comment:
I would imagine creating spans is basically free since they are very lightweight. The main thing was getting rid of the unnecessary `secure_vectors` since there malloc+free become involved.